.editorconfig

```diff
@@ -0,0 +1,28 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# newline ending every file; no trailing space
+[*]
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+# Matches multiple files with brace expansion notation
+# Set default charset
+[*.{js,json,py,xml,java,R}]
+charset = utf-8
+
+# 4 space indentation
+[*.{py,R,xml,js}]
+indent_style = space
+indent_size = 4
+
+# 2 space indentation
+[*.{java}]
+indent_style = space
+indent_size = 2
```

Comment:
I :heart: the 2-space indent.

Reply:
I can live with it, though I prefer 4 spaces. After using 2sp for ~5 years, I observed the negative impact it has even on excellent coders: the tendency to make big, complex, almighty functions with too many nested blocks of code, instead of splitting the functionality nicely. But with little discipline, this effect can be countered.